src/components/MyAccount/NewSettings/EmailForm.tsx

```diff
@@ -0,0 +1,230 @@
+    const emailRegex = /^[^@]+@[^@]+\.[^@]+$/
+    return emailRegex.test(email)
+  }
+
+  const handleInputChange = (e, index) => {
+    const { value } = e.target
+    const updatedEmails = [...tempEmails]
+    updatedEmails[index] = value
+    setTempEmails(updatedEmails)
+
+    const firstEmailEmpty = index === 0
+
+    if (firstEmailEmpty && (!value || !validateEmail(value))) {
+      setError(true)
+    } else {
+      const hasInvalidEmail = updatedEmails.some(
+        (email) => !validateEmail(email)
+      )
+      setError(hasInvalidEmail)
+    }
+  }
+
+  const handleRemoveEmail = (index) => {
+    const updatedEmails = tempEmails.filter((_, i) => i !== index)
+    setTempEmails(updatedEmails)
+
+    // Immediately revalidate remaining emails.
+    const hasInvalidEmail = updatedEmails.some((email) => !validateEmail(email))
+    setError(hasInvalidEmail)
+  }
+
+  const handleAddEmail = () => {
+    const updatedEmails = [...tempEmails, ""]
+    setTempEmails(updatedEmails)
+
+    // Immediately revalidate all emails.
+    const hasInvalidEmail = updatedEmails.some((email) => !validateEmail(email))
+    setError(hasInvalidEmail)
+  }
+
+  const handleClearableCallback = (index) => {
+    const updatedEmails = [...tempEmails]
+    updatedEmails[index] = ""
+    setTempEmails(updatedEmails)
+    setError(true)
+  }
+
+  const cancelEditing = () => {
+    setTempEmails([...emails])
+    setIsEditing(false)
+    setError(false)
+  }
+
+  const submitEmails = async () => {
+    setIsLoading(true)
+    const validEmails = tempEmails.filter((email) => validateEmail(email))
+
+    try {
+      const response = await fetch(
+        `/research/research-catalog/api/account/settings/${patronData.id}`,
+        {
+          method: "PUT",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify({ emails: validEmails }),
+        }
+      )
+
+      if (response.status === 200) {
+        await getMostUpdatedSierraAccountData()
+        setIsSuccess(true)
+        setIsEditing(false)
+      } else {
+        setIsFailure(true)
+        setTempEmails([...emails])
+      }
+    } catch (error) {
+      console.error("Error submitting emails:", error)
+    } finally {
+      setIsLoading(false)
+    }
+  }
+
+  return (
+    <>
+      {patronDataLoading || isLoading ? (
+        <SkeletonLoader contentSize={2} showImage={false} headingSize={0} />
+      ) : (
+        <Flex
+          flexDir={{ base: "column", lg: "row" }}
+          alignItems="flex-start"
+          width="100%"
+        >
+          <SettingsFormLabel icon="communicationEmail" text="Email" />
+          {isEditing ? (
+            <Flex
+              marginLeft={{ base: "l", lg: "unset" }}
+              marginTop={{ base: "xs", lg: "unset" }}
+              flexDir="column"
+              width="-webkit-fill-available"
+            >
+              {tempEmails.map((email, index) => (
```

Comment:
Can you also wrap this in a `Form` or HTML `<form>` element?
If you use the DS one, it might make it easier to have the default layout with spacing values (and means you might be able to remove the Flex components with the explicit margin values).